### PR TITLE
scripts: extract_dts_includes.py: refactor for better maintenance

### PR DESCRIPTION
**Pull request by b0661**
Refactor for better maintenance and to ease future enhancements.

This is a cutout of PR #6761 (which is a cutout of #5799) that can be applied independently.

Signed-off-by: Bobby Noelte <b0661n0e17e@gmail.com>

### CODE REVIEW

**Review by tbursztyka — APPROVED** [review]

File: scripts/dts/extract/clocks.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+# Handles:
+# - clocks
+# directives.
+#
+class DTClocks(DTDirective):
+
+    def __init__(self):
+        pass
+
+    def _extract_consumer(self, node_address, yaml, clocks, names, defs, def_label):
```
Comment (erwango) [inline scripts/dts/extract/clocks.py]: Can you remove this module in the PR?
This is new material that would deserve a dedicated review with matching yaml files and examples.

File: scripts/dts/extract/default.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    # @brief Extract directives in a default way
+    #
+    # @param node_address Address of node owning the clockxxx definition.
+    # @param yaml YAML definition for the owning node.
+    # @param prop property name
+    # @param names (unused)
+    # @param[out] defs Property definitions for each node address
+    # @param def_label Define label string of node owning the directive.
+    #
+    def extract(self, node_address, yaml, prop, names, defs, def_label):
```
Comment (erwango) [inline scripts/dts/extract/default.py]: `defs` is now global (defined in globals.py) , remove from parameters list.
Apply everywhere

**Review by erwango — CHANGES_REQUESTED** [review]

File: scripts/dts/extract/default.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+#
+class DTDefault(DTDirective):
+
+    def __init__(self):
+        pass
+
+    ##
+    # @brief Extract directives in a default way
+    #
+    # @param node_address Address of node owning the clockxxx definition.
```
Comment (erwango) [inline scripts/dts/extract/default.py]: I guess this comment is not accurate

**Review by erwango — CHANGES_REQUESTED** [review]

File: scripts/dts/extract/clocks.py (hunk 1 above)
Comment (b0661) [reply scripts/dts/extract/clocks.py]: This is the clocks directive (eg. BUS, BITS) as before.

File: scripts/dts/extract/default.py (hunk 2 above)
Comment (b0661) [reply scripts/dts/extract/default.py]: I know that defs is in globals. But the interface was kept. See current version of extract_dts_inlcudes.py. Do you really wan´t this new interface in a refcator only PR?

File: scripts/dts/extract/default.py (hunk 3 above)
Comment (b0661) [reply scripts/dts/extract/default.py]: Yes should read clocks (clockxxx is for future use).

File: scripts/dts/extract/default.py (hunk 2 above)
Comment (erwango) [reply scripts/dts/extract/default.py]: ok, my bad. This was done in my local branch bu not merged actually.
You can discard this comment

File: scripts/dts/extract/clocks.py (hunk 1 above)
Comment (erwango) [reply scripts/dts/extract/clocks.py]: Yeah, and this is actually a bit more than that as you added #clock-cells treatment.
I don't say this is not correct, but that this is more than the initial status of script features.

I've checked what was generated  and it seems true to what was generated earlier.
What puzzles me is that there is more code than in initial function that was generic to other nodes..

**Review by erwango — APPROVED** [review]

**Comment by erwango** [comment]
Approved. Let's get this merged

**Pull request merged**